exo/inference/tinygrad/models/llama.py

```
@@ -48,6 +48,56 @@ def repeat_kv(x: Tensor, n_rep: int) -> Tensor:
   return x.repeat((1, 1, 1, n_rep)).reshape(bs, seqlen, n_kv_heads*n_rep, head_dim)
 
+
+def dequantize_asymmetric(weight: Tensor, scales: Tensor, biases: Tensor, group_size: int) -> Tensor:
+  weight = weight.cast(dtype=dtypes.half).reshape(*scales.shape, group_size)  # group_size, out_features, in_features
+  scales = scales.reshape(*scales.shape, 1).expand(*scales.shape, group_size)
+  biases = biases.reshape(*biases.shape, 1).expand(*biases.shape, group_size)
+  return (weight*scales + biases).reshape(weight.shape[0], -1)
+
+
+class AffineQuantizedLinear:
+  def __init__(self, in_features: int, out_features: int, group_size: int = 64, bias: bool = False):
```

Ruff (0.8.2)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)